### ISSUE TITLE
apo:Cell is a 'Feature' not a pure 'Geometry'

### ISSUE THREAD

**Issue by dr-shorthair** [GeoscienceAustralia/AusPIX-DGGS-ontology#1]
The model for a DGGS cell shows that it has identity, type, and links to geometry(s) expressed as `geo:Geometry` so it is behaving like a Feature not a Geometry. Propose making this explicit in the ontology. 

Note that this does not affect the API implementation.

**Comment by nicholascar** [comment]
So you mean:

```turtle
aop:Cell ;
    ...
    rdfs:subClassOf geo:Feature ;
    ...
```

Not the current:

```turtle
aop:Cell ;
    ...
    rdfs:subClassOf geo:SpatialObject ;
    ...
```

My thoughts were then that you could do something like this:

```turtle
<x> a apo:Cell ;
    geo:hasGeometry "SPECIAL-DGGS-GEOMETRY"^^geox:dggsLiteral ;
```

Not just the current GeoSPARQL norm of:

```turtle
<x> a apo:Cell ;
    geo:hasGeometry \"POLYGON ((148.88888888888889 -34.663795228918765,..."^^geo:wktLiteral
```

**Comment by dr-shorthair** [comment]
> `"SPECIAL-DGGS-GEOMETRY"^^geox:dggsLiteral` 

I wonder what that would look like ... ?

**Comment by nicholascar** [comment]
Actually it should be:

```turtle
<x> a geo:Feature ;
    geo:hasGeometry "SPECIAL-DGGS-GEOMETRY"^^geox:dggsLiteral ;
```

where `SPECIAL-DGGS-GEOMETRY` is a string of integers, as per the DGGS addressing.

So *not* `aop:Cell` has a `SPECIAL-DGGS-GEOMETRY` but a standard `geo:Feature`.

**Comment by dr-shorthair** [comment]
OK. That's interesting. I had imagined an `apo:CellAggregate` class, with individuals like 
```
auspix:Agg_1
  a apo:CellAggregate ;
  dcterms:identifier "Agg_1" ;
  rdfs:member apo:R7828 ;
  rdfs:member apo:R78520 ;
  rdfs:member apo:R78521 ;
  rdfs:member apo:R78522 ;
  rdfs:member apo:R78523 ;
.
```

But that must be accompanied by a predicate to link a Feature to its equivalent from another geography (see https://github.com/CSIRO-enviro-informatics/loci-ont/issues/13).  

This is more direct, and preferable I think!
```
:F987 a geo:Feature ;
   geo:hasGeometry [
        a geo:Geometry ;
        geox:asDGGS "R7828 R78520 R78521 R78522 R78523"geox:dggsLiteral ;
     ] ;
.
```
(because `geo:hasGeometry` is an owl:ObjectProperty it must contain a blank-node or URI-reference. )

**Comment by nicholascar** [comment]
OK, I guess that's the *pure* RDF answer to this, as opposed to RDF+microformat/domain system that GeoSPARQL uses with the `wktLiteral` and other potential Literals.

Can't say which approach is better here. The *pure* RDF method might need to be tested for scaling: how many `rdf:member` individuals could a `apo:CellAggregate` have, would it be problematically large?

If a new `dggsLiteral` was made up, could a non-RDF indexer index it nicely, like Stardog's (I assume) GDAL or equivalent indexes `wktLiteral` literals for GeoSPARQL searching?

**Comment by dr-shorthair** [comment]
If you have seen GeoSPARQL with a Literal as the object of a `geo:hasGeometry` property then it is wrong. The GeoSPARQL specification and ontology are very clear that `geo:hasGeometry` is an `owl:ObjectProperty` with `rdfs:range geo:Geometry`. 

The example [shown above](https://github.com/GeoscienceAustralia/AusPIX-DGGS-ontology/issues/1#issuecomment-633363592) matches the existing patterns exactly. 
It is fine - I'd be very happy with that.

**Comment by nicholascar** [comment]
> If you have seen GeoSPARQL with a Literal as the object of a `geo:hasGeometry` property

Give me more credit than that! This is the norm, as per your example above:

```turtle
<x> 
    a geo:Feature ; 
    geo:hasGeometry [ 
        a geo:Geometry ; 
        geo:asWKT "...."^^geo:wktLiteral ;
    ] 
.
```

Still, Stardog, Allegro etc. will be using GDAL or similar to parse the WKT or GML literals so point's the same.

**Comment by dr-shorthair** [comment]
I don't understand what you were alluding to here then: 
> as opposed to RDF+microformat/domain system that GeoSPARQL uses with the wktLiteral and other potential Literals.

I've never seen anything like your [example above](https://github.com/GeoscienceAustralia/AusPIX-DGGS-ontology/issues/1#issuecomment-633358955) 
> the current GeoSPARQL norm of:
```
<x> a apo:Cell ;
    geo:hasGeometry \"POLYGON ((148.88888888888889 -34.663795228918765,..."^^geo:wktLiteral
```

**Comment by nicholascar** [comment]
Forget the example, I meant:

```turtle
<x> a apo:Cell ;
    geo:hasGeometry [
        geo:asWKT "POLYGON ((148.88888888888889 -34.663795228918765,..."^^geo:wktLiteral
    ] ;
.
```

The point is that there is RDF here connecting the `Feature` and the `Geometry` but the sup-parts of the Geometry themselves are in a microformat - here WKT, but could be DGGS details. So we have RDF + microformat and, at least in the case of WKT & GML, we have good tooling for parsing and indexing such spatial microformats outside of the RDF land but that can be used with RDF, just as Stardog does.

I think we can use outside-RDF tooling for DGGS details if recorded as literals.

**Comment by dr-shorthair** [comment]
Yeah, I was there when GeoSPARQL was being designed. My understanding of RDF &OWL was much more limited but I was always full bottle on the significance of the boundary between the semantic parts and geometry encoding. Its a different processing model.

**Comment by nicholascar** [comment]
So my fear with the approach at the top is that:

```turtle
auspix:Agg_1
  rdfs:member apo:R7828 ;
  rdfs:member apo:R78520 ;
  rdfs:member apo:R78521 ;
  rdfs:member apo:R78522 ;
  rdfs:member apo:R78523 ;
```

looks a lot less efficient than 

```turtle
auspix:Agg_1
  CONTENTS-AS-A-GEOMETRY-IN-SOME-MICROFORMAT
```

especially if the DGGS hype about speed is to be believed, and it does seem credible.

**Comment by dr-shorthair** [comment]
However, the AusPIX API http://pid.geoscience.gov.au/dggs/ausPIX/ delivers the description of a `Cell` so we also need a (Loc-I conformant) representation for the payload. 
This pattern is proposed: 
```turtle
auspix:R7852 a apo:Cell ;
  dcterms:identifier 'R7852' ;
  apo:hasNeighbourUp auspix:R7828 ;
  apo:hasNeighbourDown auspix:R7855 ;
  apo:hasNeighbourRight auspix:R8630 ;
  apo:hasNeighbourLeft auspix:R7851 ;
  geo:hasGeometry 'POLYGON ((148.88888888888889 -34.663795228918765, 150.0 -34.663795228918765, 150.0 -35.81859718358464, 148.88888888888889 -35.81859718358464))'^^geo:wktLiteral ;
  geo:sfContains auspix:R78520 , auspix:R78521 , auspix:R78522 , auspix:R78523 , auspix:R78524 , auspix:R78525 , auspix:R78526 , auspix:R78527 , auspix:R78528 ; 
  geo:sfWithin auspix:R785 ; 
  geox:centroid 'POINT (149.44444444444443 -35.23915910922053)'^^geo:wktLiteral ;
  geox:hasAreaM2 [ data:value 12971551595.0 ] ; 
.
```

So since 
```turtle
geo:hasGeometry rdfs:domain geo:Feature .
```
this entails
```turtle
apo:Cell rdfs:subClassOf geo:Feature .
```

**Comment by nicholascar** [comment]
In the DGGS Testbed, we are thinking of a `Cell` as a `Sampling Feature`. Not that LocI needs to use such a definition but conceptually, does `Cell subClassOf SamplingFeature` (of which particular ontology I don;'t know, but conceptually at least) strike you as correct @dr-shorthair?

**Comment by dr-shorthair** [comment]
The only mandatory property of `sosa:Sample` is `sosa:isSampleOf`. What type or thing does a DGGS Cell sample? Does every DGGS Cell sample one of these? 

(BTW - membership in multiple classes is fine by me, but need to understand whether this is a universal behaviour or only occasional.)

**Comment by nicholascar** [comment]
I think the thought is that since calling a service, e.g. a WFS, with a call ID will return a  lumped/averaged single point reading of *something* it's a sampling Feature. Can't say this is all though through for now but I think as the DGGS vocab is built and as your modelling work progresses, we'll get to a satisfactory set of definitions for all of this soonish.

**Comment by dr-shorthair** [comment]
Resolved by #3
